core/src/main/scala/fs2/util/Task.scala

```diff
@@ -96,18 +96,21 @@ class Task[+A](val get: Future[Either[Throwable,A]]) {
       case a => Future.now(a)
     })
 
+  /** Returns a task that performs all evaluation asynchronously. */
+  def async(implicit S: Strategy): Task[A] = Task.start(this).flatMap(identity)
+
   /**
    * Run this `Task` and block until its result is available. This will
    * throw any exceptions generated by the `Task`. To return exceptions
-   * in an `\/`, use `attemptRun`.
+   * in an `Either`, use `attemptRun`.
    */
-  def run: A = get.run match {
+  def unsafeRun: A = get.run match {
     case Left(e) => throw e
     case Right(a) => a
   }
 
   /** Like `run`, but returns exceptions as values. */
-  def attemptRun: Either[Throwable,A] =
+  def unsafeAttemptRun: Either[Throwable,A] =
```

Comment:
The general idea is that the various run methods are the only methods that *do* anything in the library. Building streams, transforming a stream to its `Free` representation, and transforming the `Free` representation in to an effectful value. All of these are referentially transparent. When using `Task`, the moment you run the task, you lose referential transparency. The `unsafe` prefix is meant as a speed bump -- something that notifies user that they are going to give up referential transparency here.